lib/synapse.rb

```diff
@@ -85,15 +76,30 @@ def reconfigure!
       @config_updated = true
     end
 
+    def available_generators
+      Hash[@config_generators.collect{|cg| [cg.name, cg]}]
```

Comment:
This method is only called during initialization of watchers. I don't really want to add state here unless I have to.